drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -32,59 +35,215 @@
-    tmp |= (AT86RF2XX_TRX_RPC_MASK__RX_RPC_EN |
-            AT86RF2XX_TRX_RPC_MASK__PDT_RPC_EN |
-            AT86RF2XX_TRX_RPC_MASK__PLL_RPC_EN |
-            AT86RF2XX_TRX_RPC_MASK__XAH_TX_RPC_EN |
-            AT86RF2XX_TRX_RPC_MASK__IPAN_RPC_EN);
-    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_RPC, tmp);
-    at86rf2xx_set_rxsensitivity(dev, RSSI_BASE_VAL);
+#if IS_USED(MODULE_AT86RF232)
+    for (int i = 0; i < num; i++) {
+        devs->base.dev_type = AT86RF2XX_DEV_TYPE_AT86RF232;
+        devs->params = *params;
+        at86rf2xx_setup((at86rf2xx_t *)devs);
+        devs++;
+        params++;
+    }
 #else
-    (void) dev;
+    (void)devs;
+    (void)params;
+    (void)num;
+#endif
+}
+
+void at86rf233_setup(at86rf233_t *devs, const at86rf233_params_t *params,
+                     uint8_t num)
+{
+#if IS_USED(MODULE_AT86RF233)
+    for (int i = 0; i < num; i++) {
+        devs->base.dev_type = AT86RF2XX_DEV_TYPE_AT86RF233;
+        devs->params = *params;
+        at86rf2xx_setup((at86rf2xx_t *)devs);
+        devs++;
+        params++;
+    }
+#else
+    (void)devs;
+    (void)params;
+    (void)num;
+#endif
+}
+
+void at86rfa1_setup(at86rfa1_t *dev)
+{
+#if IS_USED(MODULE_AT86RFA1)
+    dev->base.dev_type = AT86RF2XX_DEV_TYPE_AT86RFA1;
+    at86rf2xx_setup((at86rf2xx_t *)dev);
+#else
+    (void)dev;
+#endif
+}
+
+void at86rfr2_setup(at86rfr2_t *dev)
+{
+#if IS_USED(MODULE_AT86RFR2)
+    dev->base.dev_type = AT86RF2XX_DEV_TYPE_AT86RFR2;
+    at86rf2xx_setup((at86rf2xx_t *)dev);
+#else
+    (void)dev;
+#endif
+}
+
+size_t at86rf2xx_get_size(const at86rf2xx_t *dev)
+{
+    switch (dev->base.dev_type) {
+        default: return sizeof(at86rf2xx_t);
+#if IS_USED(MODULE_AT86RF212B)
+        case AT86RF2XX_DEV_TYPE_AT86RF212B: {
+            return sizeof(at86rf212b_t);
+        }
+#endif
+#if IS_USED(MODULE_AT86RF231)
+        case AT86RF2XX_DEV_TYPE_AT86RF231: {
+            return sizeof(at86rf231_t);
+        }
+#endif
+#if IS_USED(MODULE_AT86RF232)
+        case AT86RF2XX_DEV_TYPE_AT86RF232: {
+            return sizeof(at86rf232_t);
+        }
+#endif
+#if IS_USED(MODULE_AT86RF233)
+        case AT86RF2XX_DEV_TYPE_AT86RF233: {
+            return sizeof(at86rf233_t);
+        }
+#endif
+#if IS_USED(MODULE_AT86RFA1)
+        case AT86RF2XX_DEV_TYPE_AT86RFA1: {
+            return sizeof(at86rfa1_t);
+        }
+#endif
+#if IS_USED(MODULE_AT86RFR2)
+        case AT86RF2XX_DEV_TYPE_AT86RFR2: {
+            return sizeof(at86rfr2_t);
+        }
+#endif
+    }
+}
+
+static void at86rf2xx_disable_clock_output(at86rf2xx_t *dev)
+{
+    switch (dev->base.dev_type) {
+        default:;
+            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_CTRL_0);
+            tmp &= ~(AT86RF2XX_TRX_CTRL_0_MASK__CLKM_CTRL);
+            tmp &= ~(AT86RF2XX_TRX_CTRL_0_MASK__CLKM_SHA_SEL);
+            tmp |= (AT86RF2XX_TRX_CTRL_0_CLKM_CTRL__OFF);
+            at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_0, tmp);
```

Comment:
The variable `uint8_t tmp` needs a scope to live it. You could just enclose this all into a pair of curly braces, e.g.

```C
            {
                uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_CTRL_0);
                tmp &= ~(AT86RF2XX_TRX_CTRL_0_MASK__CLKM_CTRL);
                tmp &= ~(AT86RF2XX_TRX_CTRL_0_MASK__CLKM_SHA_SEL);
                tmp |= (AT86RF2XX_TRX_CTRL_0_CLKM_CTRL__OFF);
                at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_0, tmp);
            }
```

or you can forward declare `uint8_t tmp` before the `switch()` statement

Reply:
I am gonna take the curly braces.